react/components/ProductMessagesProgress.tsx

```diff
@@ -141,12 +148,18 @@ const ProductMessagesImport: FC<Props> = ({
           ])
         }
       })
+      .catch(() => {
+        if (lastRetry === batchProgress) {
+          setFailedTranslations([...failedTranslations, ...translations])
+        }
+        setLastRetry(batchProgress)
+      })
       .then(() => {
         if (batchProgress < batchesNumber) {
           setBatchProgress(batchProgress + 1)
         }
       })
-  }, [batchProgress, cancelled])
+  }, [batchProgress, cancelled, lastRetry])
```

Comment:
This is basically an auto retry, right? Isn't this "dangerous" if the messages system is down?

Reply:
But since it is only triggered when `lastRetry` changes, it only retries once so I think there is no problem.
Am I missing something?